packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart

```diff
@@ -139,33 +172,72 @@ class DevServerRunner {
       logger.detail('[process] process.kill()...');
       process.kill();
     }
+    _serverProcess = null;
     logger.detail('[process] killing process complete.');
   }
 
-  // TODO(renancaraujo): this method returns a future that completes when the
-  // process is killed, but it should return a future that completes when the
-  // process is finished starting.
-  /// Starts the development server.
-  Future<ExitCode> start() async {
-    var isHotReloadingEnabled = false;
+  // Internal method to cancel the watcher subscription.
+  // Make sure to call `stop` after calling this method to also stop the
+  // server process.
+  Future<void> _cancelWatcherSubscription() async {
+    if (!isWatching) {
+      return;
+    }
+    logger.detail('[watcher] cancelling subscription...');
+    await _watcherSubscription!.cancel();
+    _watcherSubscription = null;
+    logger.detail('[watcher] cancelling subscription complete.');
+  }
+
+  /// Starts the development server and a [DirectoryWatcher] subscription
+  /// that will regenerate the dev server code when files change.
+  ///
+  /// This method will throw a [DartFrogDevServerException] if called while
+  /// the dev server has been started.
+  /// This method will throw a [DartFrogDevServerException] if called after
+  /// [stop] has been called.
+  Future<void> start() async {
```

Comment:
**Nit:**

What do you think of renaming `start` and `stop` to `run` and `kill` (or a variant), similar to the API [Process](https://api.dart.dev/stable/3.0.5/dart-io/Process/kill.html) has?

I feel that `stop`  is somewhat counterintuitive since I would expect to be able to run it back again. This thought probably comes from the API [AnimationController](https://api.flutter.dev/flutter/animation/AnimationController/stop.html) has.

I feel that `kill` does hint to the reader that the server can't be _started_ again.